### DISCUSSION TITLE
Impossible to install ipa including Widget PlugIns

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#2991] [Q&A] [answered]
Hello,

I tried to create a widget on iOS using WidgetKit (SwiftUI), it's impossible to install the app when the ipa is created on Mac and Windows using AIR SDK 50.2.4.2, I reported the bug [here](https://github.com/airsdk/Adobe-Runtime-Support/issues/2989).

Using AIR SDK 50.2.3.8, I have an other problem :
I created a PlugIns directory and I copy the widgetextension.appex that I created on Xcode, I add PlugIns directory in the ipa using `adt`, the ipa is not installed on Windows and Mac I have this error :
```
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/MyApp.app/SC_Info/MyApp.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008015: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.PMgdri/extracted/Payload/MyApp.app : 0xe8008015 (A valid provisioning profile for this executable was not found.)
```

Thanks.

**Comment by pol2095** [comment]
Sorry, I have missing a parameter on the app descriptor, but `App Groups` not working on AIR SDK 50.2.4.2.

**Reply by ajwfrost** [reply]
Hi @pol2095  -> can you please elaborate on the `App Groups` thing, or raise a new issue for it? I would have thought this should be enabled in the IPA using the `Entitlements` options, you'd need to manually set this up in the app descriptor file:
https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_application-groups?language=objc
thanks

**Reply by pol2095** [reply]
The bug occurs on Windows and Mac.

using Air 50.2.3.8 and older, I add in the app descriptor :
```
<iPhone>
	<InfoAdditions>
		<![CDATA[
			<key>UIDeviceFamily</key>
			<array>
				<string>1</string>
				<string>2</string>
			</array>
			<key>CFBundleIconName</key>
			<string>AppIcon</string>
		]]>
	</InfoAdditions>
	<requestedDisplayResolution>high</requestedDisplayResolution>
	<Entitlements>
		<![CDATA[
			<key>com.apple.security.application-groups</key>
			<array>
				<string>group.com.test.MyApp.sharing</string>
			</array>
		]]>
	</Entitlements>
</iPhone>
```
just using it, it's impossible to install the app : `group.com.test.MyApp.sharing` not exist
you need to go to the Apple developer console

**step 1** : create the `App groups`

![img1](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/a2fd6cda-930a-4357-b33b-aa32bec947fb)
![img2](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/62f289ea-f4af-4b0c-b63c-48bdc191392c)
![img3](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/fb8977d1-38d1-4f6d-a34a-396fd12d8dcd)
![img4](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/be95f5cc-a0cf-4126-ac01-095697ea7ccd)

**step 2** : add the `App groups` in `com.test.MyApp`

![img5](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/5a531f1e-c07e-4851-98ae-c7bdbae0b317)
![img6](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/7909098c-9039-45e0-b117-543328bec656)
![img7](https://github.com/airsdk/Adobe-Runtime-Support/assets/6962232/84593130-8883-470e-9f9e-60bab17255c4)

**step 3** : generate an `iOS App Development` mobileprovision

using Air 50.2.3.8 and older, the app is installed correctly and using an ANE I get the `App groups` File path, but using Air 50.2.4.2, It's impossible to install the app :
```
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/MyApp.app/SC_Info/MyApp.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0x00000000: Application is missing the application-identifier entitlement.
```

I reported the bug [here](https://github.com/airsdk/Adobe-Runtime-Support/issues/2989)

**Reply by pol2095** [reply]
in `MyApp.ipa/Payload/MyApp.app/MyApp` I noticed a difference between Air 50.2.3.8 and 50.2.4.2 :
```
com.apple.security.application-groups0group.com.test.MyApp.sharing
com.apple.security.application-groups 0group.com.test.MyApp.sharing
```
I don't know if this is the cause...

**Reply by pol2095** [reply, marked as answer]
Fixed by Harman.
